### ISSUE TITLE
Add an optional flag to disable oauth authentication

### ISSUE THREAD

**Issue by AMarcedone** [google/keytransparency#603]
When testing the client, it is annoying/slows me down to have to paste the oauth link on the browser to get a token and paste it in the client. Also it makes it hard to test modifications from multiple different users.

I propose to add this flag to the kt-server (frontend.go):
```go	
fakeAuth     = flag.Bool("insecure-fake-auth", false, "INSECURE! Do not user in production! Use a fake authentication instead of getting a google oauth token from clients.")
```
which would trigger the use of FakeAuth (from core/authentication/fake.go):
```go	
var auth authentication.Authenticator
	if *fakeAuth {
		glog.Warning("INSECURE! Using fake authentication.")
		auth = authentication.NewFake()
	} else {
		var err error
		auth, err = gauth.NewGoogleAuth()
		if err != nil {
			glog.Exitf("Failed to create authentication library instance: %v", err)
		}
	}
```
Similarly, I was thinking of adding a flag to the kt-client with which the user can pick an user-id to be used in the fake authentication:

```go	
RootCmd.PersistentFlags().String("fake-auth-userid","","userid to present to the server as identity for authentication. Only succeeds if fake auth is enabled on the server side.")
```

However, I am not very familiar with grpc so not sure how to send the user-id to the server as part of the context.

Any thoughts? Does it make sense or is it better not to add such a flag for security reasons?

**Comment by gdbelvin** [comment]
Others have requested this feature as well. I think it's a good idea. 

On the outgoing call, here's how to explicitly set metadata:
https://github.com/grpc/grpc-go/blob/master/Documentation/grpc-metadata.md#sending-metadata-1 

It's probably best to adjust the fake auth to generate this for you.

**Comment by AMarcedone** [comment]
Cool. I'll prepare a PR!

**Comment by gdbelvin** [comment]
Closed via #605
